### PR TITLE
xtask: add test local flag

### PR DESCRIPTION
**Pull request by gabriele-0201**

### CODE REVIEW

**Comment by gabriele-0201** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/thrumdev/blobs/244?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `main`
  * **PR #243** <a href="https://app.graphite.dev/github/pr/thrumdev/blobs/243?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #233** <a href="https://app.graphite.dev/github/pr/thrumdev/blobs/233?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #244** <a href="https://app.graphite.dev/github/pr/thrumdev/blobs/244?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/thrumdev/blobs/244?utm_source=stack-comment).

**Review by rphmeier — APPROVED** [review]

File: xtask/src/cli.rs [hunk 1]
```diff
@@ -22,6 +22,13 @@ pub mod test {
         #[clap(long, default_value = "false")]
         pub ci: bool,
 
+        /// If the test is executed locally
+        ///
+        /// This flag will not require to add ikura-node, ikura-shim, and sov-demo-rollup
+        /// to the path. This will work ONLY if you keep the default CARGO_TARGET_DIR
+        #[clap(long, default_value = "false")]
+        pub local: bool,
```
Comment (rphmeier) [inline xtask/src/cli.rs:30]: you can use `conflicts_with = "ci"` to enforce that they are mutually exclusive at parse-time, FYI

File: xtask/src/main.rs [hunk 2]
```diff
@@ -69,6 +73,14 @@ fn init_env(ci: bool) -> anyhow::Result<()> {
         );
     }
 
+    if local {
+        let new_path = format!(
+            "{}target/release/:{}demo/sovereign/target/release:{}",
+            project_dir, project_dir, path
```
Comment (pepyakin) [inline xtask/src/main.rs:79]: why don't you use string interpolation? Seems a very good use-case for it?

**Review by pepyakin — COMMENTED** [review]
I don't understand the reasoning for this PR (would've been great to have the description filled). One may think that you can either run this tool in the CI or locally. However, it seems there are three modes: `ci`, `local` and `neither`.

Why do we need the local mode? What would be the difference with the `neither` mode.  Why is there limitation on the `CARGO_TARGET_DIR` and couldn't we workaround it (if not now, but in a follow up PR)?

File: xtask/src/cli.rs (hunk 1 above)
Comment (gabriele-0201) [reply xtask/src/cli.rs:30]: Oh, thanks! I didn't know this! I will use it for sure!

**Comment by gabriele-0201** [comment]
Yes, you're right in saying that a description would have been needed; this PR is not self-explanatory.

What I wanted to achieve with this was an automatic way to add compilation artifacts to the path without the need to source things before executing tests each time. The name 'local' does not really describe this because local development does not strictly follow the standard cargo default target directories (hence the reason for the CARGO_TARGET_DIR constraint).

I think that a feature like this would be nice, and I discovered a new possible way to do so without relying on default behavior. I will use something like `cargo metadata | jq -r '.target_directory'` for both the main workspace and the demo-rollup; this would make things shareable between each configuration. How does it seem? Please help me with the name of this flag 😅

**Comment by pepyakin** [comment]
Why do you think you need a flag for that? Why don't you do it by default, with a possibility to opt-out (a strawman: `--no-infer-bin-path`)?

And why can't this be used instead of the CI flag?

**Comment by gabriele-0201** [comment]
> why don't you do it by default, with a possibility to opt-out
Nice! I had not thought about setting it as the default option! Thanks! I thought it might be too intrusive to be as default, but in reality, it's not!

> And why can't this be used instead of the CI flag?
Yes, that was the initial idea! To remove the CI and local flags."

I will do all of this in separate pr!

**Pull request closed without merging**